shared-bindings/adafruit_bus_device/I2CDevice.c

```diff
@@ -233,9 +219,21 @@ STATIC mp_obj_t adafruit_bus_device_i2cdevice_write_then_readinto(size_t n_args,
     mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
     mp_arg_parse_all(n_args - 1, pos_args + 1, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
 
-    write(self, args[ARG_out_buffer].u_obj, args[ARG_out_start].u_int, args[ARG_out_end].u_int, false);
-
-    readinto(self, args[ARG_in_buffer].u_obj, args[ARG_in_start].u_int, args[ARG_in_end].u_int);
+    mp_obj_t dest[13];
+    mp_load_method(self->i2c, MP_QSTR_writeto_then_readfrom, dest);
+    dest[2] = mp_obj_new_int_from_ull(self->device_address);
+    dest[3] = args[ARG_out_buffer].u_obj;
+    dest[4] = args[ARG_in_buffer].u_obj;
+    dest[5] = mp_obj_new_str("out_start", 9);
+    dest[6] = mp_obj_new_int(args[ARG_out_start].u_int);
+    dest[7] = mp_obj_new_str("out_end", 7);
```

Comment:
Both busio and bitbangio default end to INT_MAX (and the normalize it to the actual length). I could check it against the buffer length at this level too but then it is a duplicated effort. Do you think it is worth checking in case a future I2C implementation does not check (which in itself is likely a bug)?